src/js/utils/Util.js

```diff
@@ -249,6 +250,30 @@ const Util = {
       search: aElement.search,
       username: aElement.username
     };
+  },
+
+  isMobile() {
+    return window.innerWidth <= viewport.MOBILE_THRESHOLD;
+  },
+
+  isViewportChangingTo(previousWidth, windowWidth) {
+    // Transition from Mobile to Desktop
+    if (
+      windowWidth > viewport.MOBILE_THRESHOLD &&
+      previousWidth < viewport.MOBILE_THRESHOLD
+    ) {
+      return viewport.DESKTOP;
+    }
+
+    // Transition from Desktop to Mobile
+    if (
+      windowWidth < viewport.MOBILE_THRESHOLD &&
+      previousWidth > viewport.MOBILE_THRESHOLD
+    ) {
+      return viewport.MOBILE;
+    }
+
+    return false;
```

Comment:
When it is indented to be reusable and part of a util, this function either should return the current viewport (`getCurrentView`) or boolean (`viewHasChanged`). But not both.
The way it is implemented now is kinda opinionated for the current usecase instead of having helpers you can use to check common things. 🤔

Reply:
I think there is a point here that we are returning different types depending on the logic, which I think makes it harder to understand this function. I would try to return the same type everywhere.